databricks_cli/dbfs/api.py

```diff
@@ -94,13 +86,21 @@ class DbfsErrorCodes(object):
 def before_sleep_on_429(retry_state):
     global time_for_last_retry
```

Comment:
Is this still needed now? This is using the global declaration. We can remove this since you're now initializing locally everytime now.

Reply:
Yes, we need it, since the function is called multiple times and `time_for_last_retry` needs to be incremented every time. Otherwise, on attempt 2 `time_for_last_retry` will be uninitialised and we'll get: `local variable 'time_for_last_retry' referenced before assignment`.

Reply:
Ah of course! The same logic still holds. Gotcha!